runtime/compiler/control/CompilationThread.cpp

```diff
@@ -8654,6 +8659,8 @@ TR::CompilationInfoPerThreadBase::wrappedCompile(J9PortLibrary *portLib, void *
 
       if (compiler)
          {
+         // The KOT needs to survive at least until we're done committing virtual guards and we must not be holding the
+         // comp monitor prior to freeing the KOT because it requires VM access.
          if (compiler->getKnownObjectTable())
             compiler->freeKnownObjectTable();
```

Comment:
Curious why are we not calling `postCompilationTasks` here instead?

Reply:
`postCompilationTasks` is relatively new; it was code that was refactored to cover a gap in the exception handling (see https://github.com/eclipse/openj9/pull/1118). As such, I refactored the code to not change the order in which things happen. It might be possible to make the call here but I would have to read the code in detail.